test/e2e/storage/drivers/csi.go

```diff
@@ -608,17 +608,18 @@ func waitForCSIDriverRegistrationOnAllNodes(driverName string, cs clientset.Inte
 		return err
 	}
 	for _, node := range nodes.Items {
-		if err := waitForCSIDriverRegistrationOnNode(node.Name, driverName, cs); err != nil {
+		if err := WaitForCSIDriverRegistrationOnNode(node.Name, driverName, cs, 5*time.Minute); err != nil {
 			return err
```

Comment:
Because previously the wait time for waitForCSIDriverRegistrationOnNode is set to 5 minutes. And the reason we are using 10 minutes in other place is that our tests are flaky occasionally because the pod get scheduled onto the node longer than 5 minutes. I don't want to change the timeout threshold for other places. Or do you think we should keep them the same?

In the flaky tests I have triaged, it can took 6-8 minutes for the registration. That's why I set the timeout to 10min. This should be long enough to get the pod running to avoid such flakiness. 

I have addressed your comment to change the wait to be ExponentialBackoff. This should keep the timeout in sync with other places.